colossalai/inference/quant/smoothquant/models/base_model.py

```diff
@@ -166,6 +127,84 @@ def prepare_inputs_for_generation(self, *args, **kwargs):
         """shortcut for model.prepare_inputs_for_generation"""
         return self.model.prepare_inputs_for_generation(*args, **kwargs)
 
+    def collect_act_scales(self, model, tokenizer, dataset_path, device, num_samples=512, seq_len=512):
+        dataset = load_dataset("json", data_files=dataset_path, split="train")
+        dataset = dataset.shuffle(seed=42)
+
+        for i in tqdm(range(num_samples)):
+            input_ids = tokenizer(
+                dataset["rows"][0][i]["row"]["text"], return_tensors="pt", max_length=seq_len, truncation=True
+            ).input_ids.to(device)
+            model(input_ids)
```

Comment:
Wonder what's the purpose of `model(input_ids)` here. Is it going to be assigned to some variable?

Reply:
Get information in  statistic function registerd in model.